tests/integ/modin/frame/test_cache_result.py

```diff
@@ -125,35 +142,135 @@ def test_cache_result_dataframe_complex_correctness(
-            snow_df.to_pandas()
-        uncached_time = perf_counter() - start
+        native_df = native_pd.concat([native_df] * 10)
+        with SqlCounter(query_count=1, union_count=299):
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
 
-        cached_snow_df = cache_and_return_df(snow_df, inplace)
+        with SqlCounter(query_count=1, union_count=29):
+            snow_df = pd.concat(
+                [pd.DataFrame([range(i, i + 5)]) for i in range(0, 150, 5)]
+            )
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
 
-        start = perf_counter()
-        with SqlCounter(query_count=1):
-            cached_snow_df = cached_snow_df.reset_index(drop=True)
+        with SqlCounter(query_count=1, union_count=9):
+            cached_snow_df = perform_chained_operations(cached_snow_df, pd)
             cached_snow_df.to_pandas()
-        cached_time = perf_counter() - start
-
-        if not inplace:
-            # If inplace, then cached_snow_df IS snow_df, so this check
-            # is meaningless.
-            with SqlCounter(query_count=1):
-                assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
-                    snow_df, native_df
-                )
 
         with SqlCounter(query_count=1):
             assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
                 cached_snow_df, native_df
             )
 
-        assert cached_time < uncached_time
+    def test_cache_result_post_pivot(self, inplace, simple_test_data):
+        pivot_kwargs = {
+            "index": "col1",
+            "columns": ["col0", "col1"],
+            "values": "col2",
+            "aggfunc": ["mean", "max"],
+        }
+        snow_df = pd.DataFrame(simple_test_data).pivot_table(**pivot_kwargs)
+        native_df = native_pd.DataFrame(simple_test_data)
+        native_df = perform_chained_operations(
+            native_df.pivot_table(**pivot_kwargs), native_pd
+        )
+        with SqlCounter(query_count=1, join_count=10, union_count=9):
+            snow_df = perform_chained_operations(snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                snow_df, native_df
+            )
+
+        with SqlCounter(query_count=1, join_count=1):
+            snow_df = pd.DataFrame(simple_test_data).pivot_table(**pivot_kwargs)
+            cached_snow_df = cache_and_return_df(snow_df, inplace)
+
+        with SqlCounter(query_count=1, union_count=9):
+            cached_snow_df = perform_chained_operations(cached_snow_df, pd)
+            assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+                cached_snow_df, native_df
+            )
+
+    def test_cache_result_post_apply(self, inplace, simple_test_data):
+        native_df = perform_chained_operations(
```

Comment:
add a comment that the additional caching isn't helping much here because apply(axis=1) already caches an intermediate result?